mongoengine/fields.py

```diff
@@ -2637,3 +2639,47 @@ def to_mongo(self, document):
             )
         else:
             return super().to_mongo(document)
+
+
+class Decimal128Field(BaseField):
+    def __init__(self, min_value=None, max_value=None, **kwargs):
+        self.min_value = min_value
+        self.max_value = max_value
+        super().__init__(**kwargs)
+
+    """
+        128-bit decimal-based floating-point field capable of emulating decimal
+        rounding with exact precision. Stores the value as a `Decimal128`
+        intended for monetary data, such as financial, tax, and scientific
+        computations.
+    """
+
+    def to_mongo(self, value):
+        if value is None:
+            return None
+        if isinstance(value, Decimal128):
+            return value
+        if isinstance(value, int):
+            return Decimal128(str(value))
+        if isinstance(value, float):
```

Comment:
combine this with the previous condition
```
if isinstance(value, (int, float)):
    return Decimal128(str(value))
```

Reply:
512062c49d8594bee860d7c440ae304f0770309e